### DISCUSSION TITLE
Merge freeze on branch

### DISCUSSION THREAD

**Discussion by sfullerbeckman** [community/community#16796] [Other Feature Feedback, Questions, & Ideas] [answered]
Many companies recurrently use code freezes or "merge freezes" on the `main` branch or a release branch to keep people from pushing/merging new code changes. Because of this, teams have had to introduce a lot of communication and process to make sure that no one merges code. Since Github does not have this feature, there is inherant risk that something might be pushed unintentionally.

It would be helpful if a merge/push freeze feature was introduced into Github core. This could be something that could be applied to a specified branch and controlled by repo admins. The freeze could have various parameters such as the length of time it should remain in place.

I think that this would be a feature that many induviduals and buisnesses would find beneficial, reducing risk and stress within teams.

**Comment by sfullerbeckman** [comment]
@davidstaheli, do you have any feedback on this?

**Reply by davidstaheli** [reply]
Hi, @sfullerbeckman. Thanks for suggesting this very important improvement. It's on our backlog and tentatively scheduled to be complete in the next 3 months (July-September).

**Reply by sfullerbeckman** [reply]
Awesome!!

**Reply by sfullerbeckman** [reply]
@davidstaheli , is this feature in a public backlog where we can see it's progess? If so, can you link me to it?

**Reply by davidstaheli** [reply]
Hi @sfullerbeckman. This isn't yet on [our public roadmap](https://github.com/github/roadmap). Updates to the roadmap are usually published the first week of each month, so it's likely this will appear in the first or second week of August. Also, right when the feature is released, a changelog will be published here: https://github.blog/changelog/.

**Reply by paescuj** [reply]
> Hi, @sfullerbeckman. Thanks for suggesting this very important improvement. It's on our backlog and tentatively scheduled to be complete in the next 3 months (July-September).

Very exciting! 🎉 One thing which I would love to see along with this is to have (simple) filter options, for example: freeze for all pull requests, expect those with a specific label.

**Reply by sfullerbeckman** [reply]
> Hi @sfullerbeckman. This isn't yet on [our public roadmap](https://github.com/github/roadmap). Updates to the roadmap are usually published the first week of each month, so it's likely this will appear in the first or second week of August. Also, right when the feature is released, a changelog will be published here: https://github.blog/changelog/.

@davidstaheli, it's September now, and it doesn't appear that this feature has been added to the public roadmap. Is it going to be soon? I just wanted to make sure that this new feature wasn't missed.

**Reply by sfullerbeckman** [reply]
Ping! It's October now. @davidstaheli, see my message above ☝🏻

**Reply by paescuj** [reply]
Is the new "[Lock branch](https://github.blog/changelog/2022-10-20-new-branch-protections-last-pusher-and-locked-branch/#lock-branch)" functionality the answer to this request?

**Reply by sfullerbeckman** [reply]
@paescuj , yes! I believe that will work.

**Comment by sfullerbeckman** [comment, marked as answer]
## Lock Branch
The new "lock branch" feature seems to do the trick! Users can now do what some refer to as a "code freeze" and prohibit any changes on a branch during a release, for example.

https://github.com/orgs/community/discussions/16796#discussioncomment-3927530
> Is the new "[Lock branch](https://github.blog/changelog/2022-10-20-new-branch-protections-last-pusher-and-locked-branch/#lock-branch)" functionality the answer to this request?

**Reply by aattalla** [reply]
But it lacks the scheduling feature. :( 

@davidstaheli is there any plan to implement a schedule that automatically turns it on or off on a weekly basis? For example, we could add a schedule feature beside this "lock branch" option to enable it on Fridays and disable it on Mondays, as an example.

**Reply by ezmirelda85** [reply]
@aattalla [Merge Freeze](https://www.mergefreeze.com/) is a [GitHub app](https://github.com/marketplace/merge-freeze) that allows you to schedule merge freezes per branch (Proactive plan). We offer a 14 day free trial and we're available to answer any questions you have!

**Reply by paescuj** [reply]
(@ezmirelda85 the first link is invalid)

**Reply by ezmirelda85** [reply]
thanks @paescuj, updated ✅
